### DISCUSSION TITLE
interactive_image without OpenCV

### DISCUSSION THREAD

**Discussion by Omnistic** [zauberzeug/nicegui#3295] [Q&A]
### Question

I have a USB3.1 FLIR camera that I want to stream in an `ui.interactive_image()`. I have read the [OpenCV webcam example](https://github.com/zauberzeug/nicegui/blob/main/examples/opencv_webcam/main.py), but I don't understand how to use it without OpenCV. I can start and stop the camera acquisition and grab single frames that I can convert to numpy arrays (I'm using the PySpin module from FLIR). So far, my understanding is that I should override the frames into a file so that this file can be used as a `source` for `ui.interactive_image()` that I can refresh with a `ui.timer()`. Am I understanding this correctly? I have a feeling that writing the files will add significant overhead. Has anyone done something like this before? Any help is appreciated, thanks.

**Comment by falkoschindler** [comment]
Hi @Omnistic,

You can also pass a `PIL_Image` to the `source` property of `ui.interactive_image`. This way you don't have to write it into a file.

**Reply by Omnistic** [reply]
Thank you @falkoschindler, I will try right away and keep you updated!
